state/cloudflare/kv/cfkv.go

```diff
@@ -0,0 +1,217 @@
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cfkv
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"reflect"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+	"golang.org/x/exp/slices"
+
+	"github.com/dapr/components-contrib/internal/component/cloudflare/workers"
+	"github.com/dapr/components-contrib/metadata"
+	"github.com/dapr/components-contrib/state"
+	"github.com/dapr/kit/logger"
+)
+
+// Link to the documentation for the component
+// TODO: Add link to docs
+const componentDocsURL = "https://TODO"
+
+// CFKV is a state store backed by Cloudflare KV.
+type CFKV struct {
+	*workers.Base
+	state.DefaultBulkStore
+	metadata componentMetadata
+}
+
+// NewCFKV returns a new CFKV.
+func NewCFKV(logger logger.Logger) state.Store {
+	q := &CFKV{
+		Base: &workers.Base{},
+	}
+	q.DefaultBulkStore = state.NewDefaultBulkStore(q)
+	q.SetLogger(logger)
+	return q
+}
+
+// Init the component.
+func (q *CFKV) Init(metadata state.Metadata) error {
+	// Decode the metadata
+	err := mapstructure.Decode(metadata.Properties, &q.metadata)
+	if err != nil {
+		return fmt.Errorf("failed to parse metadata: %w", err)
+	}
+	err = q.metadata.Validate()
+	if err != nil {
+		return fmt.Errorf("metadata is invalid: %w", err)
+	}
+	q.SetMetadata(&q.metadata.BaseMetadata)
+
+	// Init the base component
+	workerBindings := []workers.Binding{
+		{Type: "kv_namespace", Name: q.metadata.KVNamespaceID, KVNamespaceID: &q.metadata.KVNamespaceID},
+	}
+	infoResponseValidate := func(data *workers.InfoEndpointResponse) error {
+		if !slices.Contains(data.KV, q.metadata.KVNamespaceID) {
+			return fmt.Errorf("the worker is not bound to the namespace with ID '%s'; please re-deploy the worker with the correct bindings per instructions in the documentation at %s", q.metadata.KVNamespaceID, componentDocsURL)
+		}
+		return nil
+	}
+	return q.Base.Init(workerBindings, componentDocsURL, infoResponseValidate)
+}
+
+func (q *CFKV) GetComponentMetadata() map[string]string {
+	metadataStruct := componentMetadata{}
+	metadataInfo := map[string]string{}
+	metadata.GetMetadataInfoFromStructType(reflect.TypeOf(metadataStruct), &metadataInfo)
+	return metadataInfo
+}
+
+// Features returns the features supported by this state store.
+func (q CFKV) Features() []state.Feature {
+	return []state.Feature{}
+}
+
+func (q *CFKV) Delete(parentCtx context.Context, stateReq *state.DeleteRequest) error {
+	token, err := q.metadata.CreateToken()
+	if err != nil {
+		return fmt.Errorf("failed to create authorization token: %w", err)
+	}
+
+	ctx, cancel := context.WithTimeout(parentCtx, 30*time.Second)
```

Comment:
Should we make timeout duration configurable, instead of hard-coding here?

Reply:
This is during the step that's creating the worker during init, and I think hard-coding a timeout here makes sense.